### PR TITLE
CargoSpec array cleanup

### PR DESCRIPTION
**Pull request by Xaroth**
The current default CargoSpec list is very (very) verbose, making it very difficult to read what is going on. This PR tries to improve readability by doing some (albeit not-very-pretty) macro work.

This comes with some changes in sprite names and string names to make sure they all match the same format. Since I was not 100% sure on how to deal with the renaming of strings for other translations, I moved those to a separate commit, keeping only the changes to English in the main commit.

### CODE REVIEW

File: src/table/cargo_const.h [hunk 1]
```diff
@@ -7,157 +7,75 @@
...
-		STR_CARGO_PLURAL_WHEAT,          STR_CARGO_SINGULAR_WHEAT,          STR_TONS,       STR_QUANTITY_WHEAT,        STR_ABBREV_WHEAT,
-		SPR_CARGO_GRAIN,         CC_BULK,                          0x100),
-
-	MK(  6, 'MAIZ', 191, 16, 4322,  4,  40, true,  TE_NONE,
-		STR_CARGO_PLURAL_MAIZE,          STR_CARGO_SINGULAR_MAIZE,          STR_TONS,       STR_QUANTITY_MAIZE,        STR_ABBREV_MAIZE,
-		SPR_CARGO_GRAIN,         CC_BULK,                          0x100),
-
+	MK(   3, 'OIL_', 174, 16, 0x100, 4892, 25, 255,  true,       TE_NONE,           OIL,         OIL,     STR_LITERS, CC_LIQUID),
+	MK(   4, 'LVST', 208,  3, 0x100, 4322,  4,  18,  true,       TE_NONE,     LIVESTOCK,   LIVESTOCK,      STR_ITEMS, CC_PIECE_GOODS),
+	MK(   5, 'GOOD', 194,  8, 0x200, 6144,  5,  28,  true,      TE_GOODS,        GOODS,        GOODS,     STR_CRATES, CC_EXPRESS),
```
Comment (glx22) [inline src/table/cargo_const.h:61]: Bad alignment starting on this line.

**Comment by FLHerne** [comment]
Renaming `STR_*_SWEETS` and `STR_*_CANDYFLOSS` seems wrong, given that we treat `english.txt` as the definitive langfile and the strings there are "sweets" and "candyfloss". OTTD source has always been in British English, grep for 'colour'. ;-)

File: src/table/cargo_const.h [hunk 2]
```diff
@@ -7,157 +7,75 @@
...
+ * Some macros are used to automatically expand to the correct StringID consts, this
+ * means that adding/changing a cargo spec requires updating of the following strings:
+ * - STR_CARGO_PLURAL_<str_plural>
+ * - STR_CARGO_SINGULAR_<str_singular>
+ * - STR_QUANTITY_<str_plural>
+ * - STR_ABBREV_<str_plural>
+ * And the following sprite:
+ * - SPR_CARGO_<str_plural>
+ */
+#define MK(bt, label, colour, weight, mult, ip, td1, td2, freight, te, str_plural, str_singular, str_volume, classes) \
```
Comment (LordAro) [inline src/table/cargo_const.h:46]: I'd definitely consider documenting this "properly" with full doxygen treatment (`@param`)

File: src/table/cargo_const.h [hunk 3]
```diff
@@ -7,157 +7,75 @@
...
+		{bt, label, colour, colour, weight, mult, ip, {td1, td2}, freight, te, 0, 0, \
```
Comment (LordAro) [inline src/table/cargo_const.h:47]: 2 tabs indenting here is a bit weird? 1 should do

**Comment by Xaroth** [comment]
@FLHerne , @LordAro mentioned this as well on IRC. I have changed them all to prefer the english origin. This makes it a slightly bigger PR, but as you said, more in-line with the english language file.

File: src/table/cargo_const.h (hunk 1 above)
Comment (Xaroth) [reply src/table/cargo_const.h:61]: The alignment should be fixed now.

**Comment by LordAro** [comment]
You're probably not going to want to hear this, but NML could do with some renaming too...

https://github.com/OpenTTD/nml/blob/72bc37a0622c9c8fb79cb55c3e16fe9345c866b8/nml/global_constants.py#L844

Though compatibility will need to be maintained there (not that that's difficult)

File: src/table/cargo_const.h (hunk 3 above)
Comment (Xaroth) [reply src/table/cargo_const.h:47]: I based the two tab indent mainly on the original macro (which used 2 tabs) and others (build_industry.h's `MI` macro also uses two tabs). I have no preference either way so if you prefer 1 tab, that can be arranged.

Comment (LordAro) [reply src/table/cargo_const.h:47]: Eh, leave it then.

File: src/table/cargo_const.h (hunk 2 above)
Comment (Xaroth) [reply src/table/cargo_const.h:46]: Added docstrings as far as #CargoSpec  was documented. I think documenting all of CargoSpec is outside the scope of this PR.

File: src/table/cargo_const.h [hunk 4]
```diff
@@ -7,157 +7,91 @@
...
+#define MK_STR_CARGO_SINGULAR(label_singular) STR_CARGO_SINGULAR_ ## label_singular
+#define MK_STR_QUANTITY(label_plural) STR_QUANTITY_ ## label_plural
+#define MK_STR_ABBREV(label_plural) STR_ABBREV_ ## label_plural
+/** Construction macros for the #CargoSpec SpriteID entry. */
+#define MK_SPRITE(label_plural) SPR_CARGO_ ## label_plural
 
-	MK(  2, 'MAIL',  15,  4, 4550, 20,  90, false, TE_MAIL,
-		STR_CARGO_PLURAL_MAIL,           STR_CARGO_SINGULAR_MAIL,           STR_BAGS,       STR_QUANTITY_MAIL,         STR_ABBREV_MAIL,
-		SPR_CARGO_MAIL,          CC_MAIL,                          0x200),
+/*
```
Comment (LordAro) [inline src/table/cargo_const.h]: extra '*' to make it an actual doxygen comment :)

**Review by LordAro — APPROVED** [review]

**Comment by TrueBrain** [comment]
@LordAro : can this be merged before https://github.com/OpenTTD/nml/pull/176 is approved, or is it blocking on it?

**Pull request merged**